### ISSUE TITLE
DietPi-Software | Webmin: allow-hotplug interfaces not detected/handled

### ISSUE THREAD

**Issue by Sopor** [MichaIng/DietPi#2900]
#### Workaround when using Webmin to configure network:
```
sed -i 's/allow-hotplug/auto/g' /etc/network/interfaces
```
____________
I have assigned 192.168.0.10 to my RPi but still it will show something else in the `interfaces`

```
# Local
auto lo
iface lo inet loopback

# Ethernet
allow-hotplug eth0
iface eth0 inet dhcp
	address 192.168.0.100
	netmask 255.255.255.0
	gateway 192.168.0.1
	dns-nameservers 192.168.0.1
#dns-nameservers 8.8.8.8 8.8.4.4
```

As you can see here in Webmin it will show 192.168.0.10

![image](https://user-images.githubusercontent.com/5789283/59150602-3a1ed400-8a26-11e9-8dd8-e5d3f43e4512.png)

And 192.168.0.100

![image](https://user-images.githubusercontent.com/5789283/59150895-b4e9ee00-8a2a-11e9-9f11-109a0b5c8abe.png)

**Comment by MichaIng** [comment]
@Sopor 
Many thanks for your report.

Yeah it looks like Webmin is incompatible with the way DietPi creates `/etc/network/interfaces` and the other way round. What I already found in your other opened issue:
- `allow-hotplug eth0` is not detected as active interface by Webmin, even that/if it is active. Webmin expects and uses `auto eth0`, which has the issue that it will fail on boot when the cable is not connected or WiFi not available, instead of starting automatically as fast as cable connection of WiFi is found. `allow-hotplug eth0` is actually the quite common default for this reason.
  - I guess this is also the source of your doubled IP issue: Since the `allow-hotplug eth0` part for Webmin is not "active", I guess it creates a new interface somwhere which is then detected as status "Up/active" while the other one shown as "Down/not active".
-  ~Webmin re-arranges (one tab forward) the interfaces address/netmask/gateway, which breaks `dietpi-config` detection, AFAIK. The latter would be not hard to solve.~ **EDIT: Nope, works well!**

Please paste: `cat /etc/network/interfaces{,.d/*}`

**Comment by Sopor** [comment]
```
$ cat /etc/network/interfaces{,.d/*}
#/etc/network/interfaces
#Please use DietPi-Config to modify network settings.

# Local
auto lo
iface lo inet loopback

# Ethernet
allow-hotplug eth0
iface eth0 inet dhcp
        address 192.168.0.100
        netmask 255.255.255.0
        gateway 192.168.0.1
        dns-nameservers 192.168.0.1
#dns-nameservers 8.8.8.8 8.8.4.4

# Wifi
#allow-hotplug wlan0
iface wlan0 inet dhcp
address 192.168.0.100
netmask 255.255.255.0
gateway 192.168.0.1
wireless-power off
wpa-conf /etc/wpa_supplicant/wpa_supplicant.conf
#dns-nameservers 8.8.8.8 8.8.4.4
cat: '/etc/network/interfaces.d/*': No such file or directory
```

**Comment by MichaIng** [comment]
@Sopor 
Okay so the "active" interface is not added to the file at all. My next guess is now, when I see the two menus:
- "Edit Active Interface" means the changes are only applied to the current interface session, so applied most likely via `ip a` command and are not boot persistent.
- "Edit Bootup Interface" is the menu to apply changes boot persistent, thus changes here are stored to `/etc/network/interfaces`.

You can compare:
`ip a` # Should show `192.168.0.10` as currently active IP for the eth0 interface.
But after reboot, 192.168.0.100 will be applied again.
As well `ifdown eth0 && ifup eth0` should recover the persistent IP.

**Comment by Sopor** [comment]
I do use DHCP but i have it locked at 192.168.0.10 in pfsense, so don't misunderstand me here. I have not set a static IP.

**Comment by MichaIng** [comment]
@Sopor 
Ah yes sorry of course. Then
```
        address 192.168.0.100
        netmask 255.255.255.0
        gateway 192.168.0.1
        dns-nameservers 192.168.0.1
```
has no influence anyway. Strange that Webmin obviously still show this info even that it is a DHCP interface. Would make sense to hide all the "IPv4/IPv6 addresses" part when DHCP is selected above. I bet if you edit the address field to `address 192.168.0.10`, the "Edit Bootup Interface" menu will reflect his.

**Comment by diveyez** [comment]
Duplicate #2899

Both issues related to each other.

**Comment by MichaIng** [comment]
@diveyez 
Yeah, however two different issues. I leave them open until reported to Webmin.

**Comment by Sopor** [comment]
@MichaIng There is even more incompatible between Webmin and DietPi. You can check the issue https://github.com/authentic-theme/authentic-theme/issues/1339
It seems that the fake Cron job you have created make the edit_cron.cgi into a loop that will create very high CPU usage.

**Comment by MichaIng** [comment]
@Sopor 
Thanks for the hint, see my reply: https://github.com/authentic-theme/authentic-theme/issues/1339#issuecomment-501459105

Is this a Webmin issue are an authentic-theme only issue?
____________
Yeah generally having two config interfaces will mostly cause issues, as there are not always 100% defining standards and/or those are not fully met 😉. I am wondering that `cron.minutely` is even scanned since this is no GNU/Debian default but a DietPi-only implementation that we did as some user feature request. Most likely `/cron.*/*` scanning for files.

**Comment by MichaIng** [comment]
I opened an issue on Webmin GitHub page to implement correct `allow-hotplug` detection and optionally allowing to explicitly select hotplug as third interface state. For external (e.g. USB) adapters this is recommended, only for onboard adapters with static cable connection `auto` should be used: https://github.com/webmin/webmin/issues/1074

Causes this issue, most likely because `ifdown -a && ifup -a` or `systemctl restart network` is used when applying hostname settings: https://github.com/MichaIng/DietPi/issues/2899

`Waiting for user reply` = Waiting for Webmin dev thoughts about this.
